src/styles/sidebar/components/annotation-publish-control.scss

```diff
@@ -0,0 +1,119 @@
+@import './primary-action-btn';
```

Comment:
Because SASS imports are just textual inclusions and duplicate imports create duplicate CSS, all of the components should be imported just once in entry point for a SASS bundle, `sidebar/sidebase.scss` in this case. This was something I hadn't understood properly some time ago so you might find some existing incorrect `@import`s inside component .scss files.

Reply:
Trying to move this has caused a debacle :). The styles for `primary-action-btn` are in `components/primary-action-btn.scss`. That's already imported in `sidebar.scss` in the components section.

It's also used in `search-status-bar` (another component) but the only reason that works, I believe, is that `search-status-bar` is imported _after_ `primary-action-btn` (alphabetical). For `annotation-publish-control`, this doesn't work. `primary-action-btn` hasn't been imported yet so if I try to use it in that module's SCSS, ASPLODE.

Squinting at the current SASS-module organization, I am not sure where the best place is to plunk `primary-action-btn`. It obviously doesn't belong in `components` as it isn't one and needs to be shared, but I dunno the best place to stick it.

Reply:
> Trying to move this has caused a debacle :). 

`primary-action-btn.scss` defines both mixins and classes. In `annotation-publish-control.scss` you only need the mixin. Extract the mixins into a separate module and import that at the top of `sidebar.scss` where the other shared mixins are imported. You should then be able to keep `components/` imports sorted in alphabetical order.

> It's also used in search-status-bar (another component)

Hmm, `search-status-bar.scss` doesn't reference anything from `primary-action-btn.scss`. The `search-status-bar.html` _template_ does, but that's not used to build the CSS bundle.

> It obviously doesn't belong in components

I look at it as a CSS-only component, for cases when all you need is to apply some basic styling to a standard HTML element. It isn't 100% clear to me whether we'll still have a use for many of those by the end of the conversion, or whether we'll just create small components for everything. It's fine to put it in `sidebar/components/<name>.scss`.